Stratis.Bitcoin/Features/BlockStore/BlockStoreLoop.cs

```diff
@@ -1,347 +1,187 @@
-﻿using System;
-using System.Collections.Concurrent;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading;
-using System.Threading.Tasks;
-using Microsoft.Extensions.Logging;
+﻿using Microsoft.Extensions.Logging;
 using NBitcoin;
 using Stratis.Bitcoin.Base;
 using Stratis.Bitcoin.BlockPulling;
+using Stratis.Bitcoin.BlockStore;
 using Stratis.Bitcoin.Common.Hosting;
 using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Features.BlockStore.LoopSteps;
 using Stratis.Bitcoin.Utilities;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace Stratis.Bitcoin.Features.BlockStore
 {
-	public class BlockStoreLoop
-	{
-		private readonly ConcurrentChain chain;
-		public BlockRepository BlockRepository { get; } // public for testing
-		private readonly NodeSettings nodeArgs;
-		private readonly StoreBlockPuller blockPuller;
+    public sealed class BlockStoreLoop
+    {
+        internal readonly ConcurrentChain Chain;
+        public BlockRepository BlockRepository { get; }
+        private readonly NodeSettings nodeArgs;
+        internal StoreBlockPuller BlockPuller { get; private set; }
         private readonly BlockStoreCache blockStoreCache;
-	    private readonly INodeLifetime nodeLifetime;
-	    private readonly IAsyncLoopFactory asyncLoopFactory;
-	    private readonly BlockStoreStats blockStoreStats;
-	    private readonly ILogger storeLogger;
+        private readonly INodeLifetime nodeLifetime;
+        private readonly IAsyncLoopFactory asyncLoopFactory;
+        private readonly BlockStoreStats blockStoreStats;
+        private readonly ILogger storeLogger;
 
         public ChainState ChainState { get; }
-		public ConcurrentDictionary<uint256, BlockPair> PendingStorage { get; }
-	    public ChainedBlock StoredBlock { get; private set; }
-
-        public BlockStoreLoop(ConcurrentChain chain, 
-            BlockRepository blockRepository, 
-            NodeSettings nodeArgs,
-			ChainState chainState,
-			StoreBlockPuller blockPuller,
+        public ConcurrentDictionary<uint256, BlockPair> PendingStorage { get; }
+        public ChainedBlock StoredBlock { get; internal set; }
```

Comment:
Rename this to Tip or StoreTip
And maybe add a new method called .SetStoreTip(ChainedBlock )
It will also set the ChainState.HighestPersistedBlock